src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1574,13 +1594,23 @@ public function getParameterBag(): ParameterBagInterface
             foreach ($this->buildParameters as $name => $value) {
                 $parameters[$name] = $value;
             }
-            $this->parameterBag = new FrozenParameterBag($parameters);
+            $this->parameterBag = new FrozenParameterBag($parameters, $this->deprecatedParameters ?? []);
```

Comment:
why `??[]` ? We should always have an array IMO.

Reply:
Because as you noticed in your previous comment the property may not be dumped at all.

Reply:
we know at dump-time whether the property is going to exist, so let's remove this when not

Reply:
Fixed.